### ISSUE TITLE
profile=hiking-mountain routes over way with foot=no

### ISSUE THREAD

**Issue by vincentius63** [abrensch/brouter#604]
Hi,

http://www.brouter.de/brouter-web/#map=16/49.4824/8.4732/osm-mapnik-german_style&lonlats=8.469056,49.480667;8.466704,49.47854&profile=hiking-mountain

brouter-web routes through the "Tunnelstraße". This way is currently set with access foot=no because of construction works https://www.openstreetmap.org/way/114399554#map=17/49.47835/8.46851

Most bike profiles do it right: http://www.brouter.de/brouter-web/#map=16/49.4824/8.4732/osm-mapnik-german_style&lonlats=8.469056,49.480667;8.466704,49.47854

Best regards
Vinzenz

**Comment by poutnikl** [comment]
It belongs rather here:  https://github.com/poutnikl/Brouter-profiles/issues
I have noticed I have set foot access right, but then decided if there is no footaccess but is there bikeaccess, it falls back to be allowed with penalty 4, being taken, if the detour is 5x times longer.

While it affect all foot profiles, it is rather funny you have used hiking mountain profile for city walking.

**Comment by quaelnix** [comment]
@poutnikl, the problem is the `issidewalk` logic.

**Comment by poutnikl** [comment]
Yes, but that is tricky, as foot=no may apply on the main road, while there is a sidewalk.

**Comment by poutnikl** [comment]
> @poutnikl, the problem is the `issidewalk` logic.

Hmm, you seems right.

as workaround intil fixed, you can modify and manually upload the profile to the BRouter server with modification

from
`
assign issidewalk   sidewalk=left|right|both|yes`

to

`assign issidewalk   and not foot=no  sidewalk=left|right|both|yes       `

**Comment by quaelnix** [comment]
I would suggest that we don't fix this bug because, as you said, it would very likely do more harm than good.

**Comment by poutnikl** [comment]
I quite agree with you.  there will be always false positives (routing via denied roads) and false negatives (avoiding available roads). One then has to evaluate pros and cons of "allow/deny always".

**Comment by vincentius63** [comment]
> While it affect all foot profiles, it is rather funny you have used hiking mountain profile for city walking.

It's the only walking profile in [brouter-web](https://brouter.de/brouter-web/). Because of that I used this as this issue came up in https://community.openstreetmap.org/t/fussgangerrouting-trotz-foot-no/101846

**Comment by poutnikl** [comment]
I see. I seldom use Brouter web, usually only for convenient profile testing during development of my profile templates. Note that all my [foot profiles](https://github.com/poutnikl/Brouter-profiles/tree/master/HikeProfiles) are based on the [single template](https://github.com/poutnikl/Hiking-Poutnik) so all behave the same way here.

There is suspicion that denying access for proerly tagged foot=no for road under construction will deny access for road in properly tagged foot=no with denied access for walking on the main road only, while there are sidewalking structures.

Bike profiles do not have this issue for obvious reasons.

**Comment by quaelnix** [comment]
@poutnikl, what if we would replace:
```
or issidewalk
```
with:
```
or and issidewalk not access=no
```
That would probably almost exclusively catch ways closed due to construction work.

**Comment by quaelnix** [comment]
@poutnikl, what if we would replace:
```
or issidewalk
```
with:
```
or and issidewalk not access=no
```
This would probably almost exclusively affect ways that are closed for construction work.

**Comment by quaelnix** [comment]
Unfortunately it's not that simple. To make it really work, we would need something ugly like this:
```
or and issidewalk not ( and access=no and ( or bicycle= bicycle=no ) and ( or motor_vehicle= motor_vehicle=no ) and ( or bus= bus=no ) ( or psv= psv=no ) )
```
This would fix ~ 90 % of all the cases in europe: https://overpass-turbo.eu/s/1ymG

**Comment by poutnikl** [comment]
Or
assign issidewalk   if  (sidewalk=left|right|both|yes) 
   then ( not ( and access=no and ( or bicycle= bicycle=no ) and ( or motor_vehicle= motor_vehicle=no ) and ( or bus= bus=no ) ( or psv= psv=no ) )  ) 
   else false

**Comment by vincentius63** [comment]
Thank you for your great work. I feared that it was not easy.

Best regards
